src/ape/managers/chain.py

```diff
@@ -359,9 +360,22 @@ def base_fee(self) -> int:
     def pending_timestamp(self) -> int:
         """
         The current epoch time of the chain, as an ``int``.
+        Set the timestamp to fast-forward time into the future.
+
+        Usage example::
+
+            from ape import chain
+
+            chain.pending_timestamp += 3600
         """
 
-        return int(time.time())
+        return int(time.time() + self._time_offset)
+
+    @pending_timestamp.setter
+    def pending_timestamp(self, new_value: int):
+        provider = self._get_test_provider(TestProviderAPI.fast_forward_time.__name__)
```

Comment:
What is going on here? Can't this be `self.provider`?

Reply:
The `_get_test_provider` helper private method just ensures that `self.provider` is indeed a `TestProviderAPI` implementation. The `TestProviderAPI` is the mixin that has the snapshotting and other development-only related methods that normal `ProviderAPI` implementations won't have.

Reply:
Oh also, `mypy` is happier when we know it is a `TestProviderAPI`, that is kind of what drove this

Reply:
outside the scope of this PR, but really don't like how this works

down below, I see that the mechanics are working the way I expected, but was hoping somehow it would be a lot more ergonomic rather than having to pass `TestProviderAPI.fast_forward_time.__name__` to it...

maybe this can be hacked by wrapping `self.provider.fast_forward_time` in a `try..except` that catches `AttributeError` and converts to `NotImplementedError` instead e.g.:

```py
@pending_timestamp.setter
    def pending_timestamp(self, new_value: int):
        try:
            self.provider.fast_forward_time(new_value)
        except AttributeError as e:
            raise NotImplementedError from e
```

It would be even neater if fields that are in `TestProviderAPI` but not `ProviderAPI` would instead just raise this error directly so we don't have to hack it so much e.g.:

```py
@pending_timestamp.setter
    def pending_timestamp(self, new_value: int):
        self.provider.fast_forward_time(new_value)
        # raises `NotImplementedError` if only `ProviderAPI`
```

Reply:
This might be inside the scope of this PR if this is the only instance where this occurs :thinking:

Reply:
Ah ha! Okay, maybe we solve it like this:

1. All fields/methods are in `ProviderAPI` that are destined for `TestProviderAPI`
2. Those fields all raise `NotImplementedError` in `ProviderAPI`
3. Override those fields in `TestProviderAPI` to instead be decorated with `@apimethod`

Now you can initialize normal `ProviderAPI` classes without issue, but if a `TestProviderAPI` subclass does not implement the methods, it will throw. Most importantly, mypy is happy lol.

Reply:
NOTE: Come back to this later and fix